test/Autofac.Test/Concurrency/ConcurrencyTests.cs

```diff
@@ -132,6 +133,20 @@ public void WhenTwoThreadsInitialiseASharedInstanceSimultaneouslyViaChildLifetim
             Assert.Single(results.Distinct());
         }
 
+        [Fact]
+        public void WhenTwoThreadsResolvesNotAlreadyRegisteredType_DoesNotThrow()
+        {
+            for (int i = 0; i < 10000; i++)
+            {
+                var builder = new ContainerBuilder();
+                builder.RegisterSource(new AnyConcreteTypeNotAlreadyRegisteredSource());
+                var container = builder.Build();
+                Parallel.Invoke(
```

Comment:
maybe it's better to ad more `() => container.Resolve<A>()`  to fail with higher probability?

Reply:
done